File: swift/Tests/CoinTypeConfigurationTests.swift
// SPDX-License-Identifier: Apache-2.0
//
// Copyright © 2017 Trust Wallet.

import XCTest
import WalletCore

class CoinTypeConfigurationTests: XCTestCase {
    func testNativeTokenNameDiffersFromNameForSmartChain() {
        let name = CoinTypeConfiguration.getName(type: .smartChain)
        let nativeTokenName = CoinTypeConfiguration.getNativeTokenName(type: .smartChain)

        XCTAssertEqual(name, "BNB Smart Chain")
        XCTAssertEqual(nativeTokenName, "BNB")
        XCTAssertNotEqual(name, nativeTokenName)
    }

    func testNativeTokenNameMatchesNameForBitcoin() {
        let name = CoinTypeConfiguration.getName(type: .bitcoin)
        let nativeTokenName = CoinTypeConfiguration.getNativeTokenName(type: .bitcoin)

        XCTAssertEqual(name, nativeTokenName)
    }
}

File: kotlin/wallet-core-kotlin/src/commonTest/kotlin/com/trustwallet/core/test/CoinTypeConfigurationTests.kt
package com.trustwallet.core.test

import com.trustwallet.core.CoinType
import com.trustwallet.core.CoinTypeConfiguration
import com.trustwallet.core.LibLoader
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals

class CoinTypeConfigurationTests {

    init {
        LibLoader.loadLibrary()
    }

    @Test
    fun testNativeTokenNameDiffersFromNameForSmartChain() {
        val name = CoinTypeConfiguration.getName(CoinType.SmartChain)
        val nativeTokenName = CoinTypeConfiguration.getNativeTokenName(CoinType.SmartChain)

        assertEquals("BNB Smart Chain", name)
        assertEquals("BNB", nativeTokenName)
        assertNotEquals(name, nativeTokenName)
    }

    @Test
    fun testNativeTokenNameMatchesNameForBitcoin() {
        val name = CoinTypeConfiguration.getName(CoinType.Bitcoin)
        val nativeTokenName = CoinTypeConfiguration.getNativeTokenName(CoinType.Bitcoin)

        assertEquals(name, nativeTokenName)
    }
}
